server/Models/order.js

```diff
@@ -0,0 +1,32 @@
+//pizza type and quantity 
+
+//example
+// 30 bags of rice
+
+///ORDER: id, [list of order_item represented by order_item_id ]
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+const { ObjectId } = Schema;
+const DbInstanceConnections = require('../Database/database')
+
+const dbSingleton = new DbInstanceConnections().getInstance();
+const db = mongoose.connection;
+
+db.on('error', console.error.bond(console, 'Order Model connection error'))
+
+db.once('open' , () => {

```

Comment:
Is not this connection already open through the singleton?

Reply:
Not really, the Singleton only instantiates a new database in case none has been instantiated.

What the `db.once` does is:
-  The event handler is connected to the connection returned by  `db = mongoose.connection`
- Then It will be triggered one time for the first open event that happens